compiler/test/fail_compilation/test23438.d

```diff
@@ -0,0 +1,45 @@
+/* REQUIRED_ARGS: -preview=dip1000
+TEST_OUTPUT:
+---
+fail_compilation/test23438.d(114): Error: scope variable `x` assigned to global variable `escaped`
+---
+*/
+
+// https://github.com/dlang/dmd/pull/14601
+
+#line 100
+
+int global;
+int* escaped;
+
+void quxfail() @safe
+{
+    int stack=1337;
+    int* foo(return scope int* x) @safe
+    {
+        int* bar(return scope int* y) @safe
+        {
+            return x;
+        }
+        auto p = &bar;
+        escaped = bar(&global); // fail
```

Comment:
I find the error confusing. There is no `x` at this line. The correct error should be at line 22 (please stop using #line as it makes it difficult to review the test) where `x` is returned from `bar` because that is what the user needs to modify.

Reply:
I use #line because it enables adding/removing lines without having to change the TEST_OUTPUT lines.

Reply:
The code at line 22 is allowed. It's 25 where the error is, as it's using bar() to return x.

Reply:
Sorry but I do not agree. The problem is that `bar` is returning a `scope` variable. `bar` gets a context pointer which points to 2 scope variables (`x` and `stack`). Trying to return a reference to any of those should be an error. Also, consider that function `bar` may have more code, including branches:

```d
  int* bar(return scope int* y) @safe                                  
  {
      if (1)  
          return escaped;
      else  if (...)
          return x;
      else if (...) { }
  }
```

Still, when analyzing `escape = bar(&global)` the error is "scope variable `x` assigned to global variable `escaped`" . That is false and misleading in this case.

Reply:
I think we'll have to agree to disagree.

Reply:
It doesn't matter that the `return x` is unreachable. It is reachable according to the language semantics. The error message isn't wrong - the code is.